src/coldfront_plugin_api/scim_v2/adapters.py

```diff
@@ -0,0 +1,61 @@
+'''
+Defines the SCIM adapter for coldfront users and groups
+'''
+
+from django_scim.adapters import SCIMUser, SCIMGroup
+from django_scim import constants
+
+from coldfront_plugin_api import auth, utils
+
+class SCIMColdfrontUser(SCIMUser):
+    id_field = "username"
+
+    def to_dict(self):
+
+        d = {
+            'schemas': [constants.SchemaURI.USER],
+            "id": self.obj.username,
+            "externalId": self.obj.username,
+            "userName": self.obj.username,
+            "name": {
+                "givenName": self.obj.first_name,
+                "familyName": self.obj.last_name,
+            },
+            "emails": [
+                {
+                    "value": self.obj.email,
+                    "type": "work",
+                    "primary": True,
+                }
+            ],
+            "meta": {
+                "resourceType": "User",
+                "created": self.obj.date_joined.isoformat(),
+            },
+        }
+
+        return d
+
+    def from_dict(self, d):
+
+        '''
+        Overide from SCIMUser, this function first check if user exists through one of 
+        the configured search providers, and if yes, will fetch additional information from there.
+
+        Otherwise, for setting the attributes manually from the request's payload
+        '''
+        username = d.get('userName')
+        found = utils.find_user(username) # Fetch user
```

Comment:
We're using `d.get('username')`, which means that `username` could be `None`. Should we be calling `utils.find_user` in that case?

Reply:
I've override the `validate_dict` function, which will check if the username field exists